### PR TITLE
Setting up weasel branch

### PR DESCRIPTION
**Pull request by svlandeg**
## Description
Setting up the `weasel` branch to coordinate migration of `spacy projects` to `weasel`.
The functionality of this PR accompanies https://github.com/explosion/weasel/pull/9 and https://github.com/explosion/weasel/pull/11.

In a nutshell, we've moved all project-related functionality & tests verbatim to `weasel`. The big exception are the projects tests in `test_cli_app.py` - those remain and will be fixed in follow up PRs to the `explosion:feature/weasel` branch, as we still want `python -m spacy projects` to work exactly as before.

### Types of change
WIP - migration

## Checklist
The test might just fail right now, that's OK. We're not merging this into `master`, just into a specific feature branch to continue working on. Specifically, the ones in `test_cli_app.py` will fail right now.

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [ ] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/util.py [hunk 1]
```diff
@@ -949,130 +942,6 @@ def replace_model_node(model: Model, target: Model, replacement: Model) -> None:
...
-    current = Path(path).resolve()
-    os.chdir(str(current))
-    try:
-        yield current
-    finally:
-        os.chdir(str(prev_cwd))
-
-
-@contextmanager
-def make_tempdir() -> Generator[Path, None, None]:
```
Comment (adrianeboyd) [inline spacy/util.py:1046]: I'm pretty sure we still need `make_tempdir` for the test suite. I guess you could move it into `spacy.tests.util`, but it seems fine to leave it here?

Comment (svlandeg) [reply spacy/util.py:1046]: There's a few more things that I need to fix, I'll take it out of review when I'm done!

File: spacy/cli/_util.py [hunk 2]
```diff
@@ -135,115 +131,6 @@ def _parse_override(value: Any) -> Any:
...
-            if step not in command_names:
-                msg.fail(
-                    f"Unknown command specified in workflow '{workflow_name}': {step}",
-                    f"Workflows can only refer to commands defined in the 'commands' "
-                    f"section of the {PROJECT_FILE}.",
-                    exits=1,
-                )
-
-
 def get_hash(data, exclude: Iterable[str] = tuple()) -> str:
```
[the remark this answers is not in the document]
Comment (bdura) [reply spacy/cli/_util.py:134]: Ok, I looked at it - I think `get_hash` and `get_checksum` can go. `show_validation_error` needs to remain for now, although the next step would be to import it from weasel.

File: spacy/cli/_util.py [hunk 3]
```diff
@@ -381,119 +268,10 @@ def ensure_pathy(path):
...
-    cmd = f"git ls-remote {repo} {branch}"
-    # We might be tempted to use `--exit-code` with `git ls-remote`, but
-    # `run_command` handles the `returncode` for us, so we'll rely on
-    # the fact that stdout returns '' if the requested branch doesn't exist
-    ret = run_command(cmd, capture=True)
-    exists = ret.stdout != ""
-    return exists
-
-
 def get_git_version(
```
Comment (bdura) [inline spacy/cli/_util.py:271]: This is only referenced in the tests.

File: spacy/cli/_util.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bdura) [reply spacy/cli/_util.py:134]: The following functions are only referenced within the tests or never used:
- `upload_file`
- `download_file`
- `ensure_pathy`

File: spacy/cli/download.py [hunk 4]
```diff
@@ -19,7 +19,7 @@ def download_cli(
     # fmt: off
     ctx: typer.Context,
     model: str = Arg(..., help="Name of pipeline package to download"),
-    direct: bool = Opt(False, "--direct", "-d", "-D", help="Force direct download of name + version"),
+    direct: bool = Opt(False, "--direct", "-d", "-D", help="Force direct download PROJECT_of name + version"),
```
Comment (bdura) [inline spacy/cli/download.py:22]: Is this a typo?

File: spacy/errors.py [hunk 5]
```diff
@@ -828,7 +826,6 @@ class Errors(metaclass=ErrorsWithCodes):
             "assign it a name, e.g. `@Language.component('your_name')`. You can "
             "then run `nlp.replace_pipe('{name}', 'your_name')`.")
     E969 = ("Expected string values for field '{field}', but received {types} instead. ")
-    E970 = ("Can not execute command '{str_command}'. Do you have '{tool}' installed?")
```
Comment (bdura) [inline spacy/errors.py:831]: This is still in use : https://github.com/explosion/spaCy/pull/12456/files#diff-edf9a5728c2b543e15658f4ab1e7ca86a63b0f18d3751b3e0c08c8f9a147083bR997

**Review by bdura — COMMENTED** [review]
There are still some functions that can be safely removed, since they are now only referenced within the tests.

File: spacy/cli/_util.py (hunk 3 above)
Comment (svlandeg) [reply spacy/cli/_util.py:191]: Yes, but it's referenced from `test_cli_app.py` which should remain in spaCy as is, to ensure `spacy project ...` still works, right?

File: spacy/cli/_util.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (svlandeg) [reply spacy/cli/_util.py]: Thanks for the detailed review!

File: spacy/cli/download.py (hunk 4 above)
Comment (svlandeg) [reply spacy/cli/download.py]: Looks like it!

**Pull request merged**